python/ray/autoscaler/util.py

```diff
@@ -85,6 +105,15 @@ def fillout_defaults(config: Dict[str, Any]) -> Dict[str, Any]:
     return defaults
 
 
+def merge_node_type_config(config: Dict[str, Any]) -> Dict[str, Any]:
+    if "head_node_type" in config:
+        config["head_node"] = copy.deepcopy(
```

Comment:
Should raise a better message on bad key

Reply:
Maybe it's worth checking for defaults in `validate_config`?